src/main/java/com/warrenstrange/googleauth/GoogleAuthenticator.java

```diff
@@ -166,10 +168,44 @@ public GoogleAuthenticator(GoogleAuthenticatorConfig config)
         }
 
         this.config = config;
+
+        this.secureRandom = new ReseedingSecureRandom(
+                getRandomNumberAlgorithm(),
+                getRandomNumberAlgorithmProvider()
+        );
+    }
+
+    public GoogleAuthenticator(final String randomNumberAlgorithm, final String randomNumberAlgorithmProvider)
+    {
+        this(new GoogleAuthenticatorConfig(), randomNumberAlgorithm, randomNumberAlgorithmProvider);
+
+    }
+
+    public GoogleAuthenticator(GoogleAuthenticatorConfig config, final String randomNumberAlgorithm, final String randomNumberAlgorithmProvider)
+    {
+        if (config == null)
+        {
+            throw new IllegalArgumentException("Configuration cannot be null.");
+        }
+
+        this.config = config;
+
+        if (randomNumberAlgorithm == null && randomNumberAlgorithmProvider == null)
```

Comment:
It would be nice to update the javadoc for these parameters, especially since they allow null and don't use `DEFAULT_RANDOM_NUMBER_ALGORITHM` or `DEFAULT_RANDOM_NUMBER_ALGORITHM_PROVIDER `.

Reply:
https://github.com/wstrange/GoogleAuth/pull/87/commits/98274b6e6b8835353b00ee93d2f0c174798a1035